tests/test_main.py

```diff
@@ -0,0 +1,38 @@
+import requests
+from fastapi.testclient import TestClient
+from main import app
+
+
+client = TestClient(app)
+
+
+# check that the homepage returns a successful status code
+def test_app():
+    resp = client.get("/")
+    assert resp.raise_for_status() != requests.exceptions.HTTPError
+
+
+# check that "/facts" works as it should
+def test_get_cat_facts():
+
+    resp = client.get("/facts")
+
+    # check that get_cat_facts()'s response is successful
+    assert resp.raise_for_status() != requests.exceptions.HTTPError
+
+    # check that the "data" portion is returning a cat fact
+    resp = resp.json()
+    assert list(resp.keys()) == ["fact"]
```

Comment:
Same here, you need to test that your API is returning what is expected to return. in this case I wrote a test example, that returns a single fact, so random always returns the same one that your API received from the cat facts service.
```
@mock.patch("main.requests.get")
def test_get_cat_facts(mock_requests_get):
    mock_data = {'fact': 'Cats sleep for 16 hours a day.'}
    mock_response = mock.Mock()
    mock_response.status_code = 200
    mock_response.json.return_value = {
        "data": [mock_data]
    }
    mock_requests_get.return_value = mock_response
    resp = client.get("/facts")

    assert resp.status_code == 200
    assert resp.json() == mock_data
```